Assets/Scripts/InputManager/InputManager.cs

```diff
@@ -9,30 +9,40 @@ namespace Inputs
 {
     public class InputManager : Singleton<InputManager>
     {
-
-        public PlayerController playerController { get; set; }
-
-        private float horizontalVal, verticalVal;
+        private List<InputComponent> inputComponentList = new List<InputComponent>();
 
         // Update is called once per frame
         void Update()
         {
-            horizontalVal = Input.GetAxis("Horizontal1");
-            verticalVal = Input.GetAxis("Vertical1");
-
-            if (horizontalVal != 0 || verticalVal != 0)
+            foreach (InputComponent inputComponent in inputComponentList)
             {
-                if (playerController != null)
+                inputComponent.OnUpdate();
+
+                if (inputComponent.horizontalVal != 0 || inputComponent.verticalVal != 0)
```

Comment:
let's refactor this to be inside the input component - the InputManager does not need to know about the controller etc